### ISSUE TITLE
AutoEstimator should not always get metric mode from Evaluator

### ISSUE THREAD

**Issue by yushan111** [intel-analytics/analytics-zoo#4023]
The metric in `AutoEstimator.fit` is the metric name for optimization in search algorithm and scheduler. It should be one of the metric names to evaluate pytorch/keras model in each trial.

Currently we always use the built-in `Evaluator` to evaluate pytorch model and we use the metrics in `model.compile` in user-customized `model_creator` to evaluate keras model.

Therefore, it is ok to get metric mode from `Evaluator` for `AutoEstimator.from_pytorch`, but it might not work right for other model (tf.keras, xgboost) with their built-in metric names.
